src/lib/containers/TotalQueryResults.tsx

```diff
@@ -188,47 +199,89 @@ const TotalQueryResults: FunctionComponent<TotalQueryResultsProps> = ({
     calculateTotal()
   }, [parentLoading, accessToken, lastQueryRequest])
 
-  const removeFacetSelection = ({
-    facet,
-    selectedValue,
-  }: FacetWithSelection) => {
+  const removeFacetSelection = (facetWithSelection: FacetWithSelection) => {
+    const { facet, selectedValue } = facetWithSelection
     if (facet.facetType === 'enumeration') {
       applyChangesToValuesColumn(
         lastQueryRequest,
         facet as FacetColumnResultValues,
-        applyChanges!,
+        applyChanges,
         selectedValue?.value,
         false,
       )
     } else {
       applyChangesToRangeColumn(
         lastQueryRequest,
         facet as FacetColumnResultRange,
-        applyChanges!,
+        applyChanges,
         [RadioValuesEnum.ANY, RadioValuesEnum.ANY],
       )
     }
   }
 
-  const removeSearchQuerySelection = (columnName: string, value: string) => {
+  const removeColumnSearchQuerySelection = (
+    columnName: string,
+    value: string,
+  ) => {
     const cloneLastQueryRequest = cloneDeep(lastQueryRequest)
     if (!cloneLastQueryRequest.query.additionalFilters) {
       return
     }
-    cloneLastQueryRequest.query.additionalFilters = cloneLastQueryRequest.query.additionalFilters
-      ?.map(el => {
-        return {
-          columnName: el.columnName,
-          values:
-            el.columnName === columnName
-              ? el.values.filter(el => el !== value)
-              : el.values,
-          operator: el.operator,
-          function: el.function,
-          concreteType: el.concreteType
+    cloneLastQueryRequest.query.additionalFilters =
+      cloneLastQueryRequest.query.additionalFilters?.reduce(function (
```

Comment:
Big style question--should we use `reduce()` or `map().filter()`?

Would need to do something like

```ts
cloneLastQueryRequest.query.additionalFilters?.map(qf => {
	if (isTextMatchesQueryFilter(qf) {
		return qf
	} else if (isColumnSingleValueQueryFilter(el) || isColumnMultiValueFunctionQueryFilter(el)) {
		// transform by removing matching column/value tuple
		// ...
		return transformedQueryFilter
	}
}).filter(queryFilter => {
	if (isTextMatchesQueryFilter(qf)) {
		return true
	} else if (isColumnSingleValueQueryFilter(el) || isColumnMultiValueFunctionQueryFilter(el)) {
		return values.length > 0
	}
})
```

Reply:
I think reduce is easier to understand in this case, but I don't have a strong opinion on this.
@conundrumgirl ?

Reply:
In both map and filter we have the same conditions check. So if we could get rid of this redundancy by using reduce I think it would be preferable